src/viewProviders/resources.ts

```diff
@@ -109,7 +108,12 @@ export class ResourceViewProvider implements vscode.TreeDataProvider<ResourceVie
     } else {
       // --- ROOT-LEVEL ITEMS ---
       // NOTE: we end up here when the tree is first loaded
-      return await loadResources(this.forceDeepRefresh);
+      const toplevelItems = await loadResources(this.forceDeepRefresh);
+      if (this.forceDeepRefresh) {
```

Comment:
Damned if do, damned if don't. Half the people like style of 'always clear the flag even if wasn't set in first place' in argument of 'is faster and fewer LOC', but other half like the explicitness. I erred on the latter this time given isn't anywhere close to a high performance inner loop needing to avoid branch prediction misses, and also gives a good place for the comment tying off the deep refresh flow. YMMV, OC.